types/shares.go

```diff
@@ -184,3 +192,327 @@ func zeroPadIfNecessary(share []byte, width int) []byte {
 	}
 	return share
 }
+
+// /////////////////////////////
+// Merging
+// ////////////////////////////
```

Comment:
IMO, we should split out the encoding, or rather the merge-split logic into separate files, e.g. share_split.go, shares_merge.go. Or, shares_encoding.go if we wanted both directions in one file.

Reply:
good idea! https://github.com/lazyledger/lazyledger-core/pull/261/commits/1cb40309ed9a65b5c18b8fc8c38491c0a0fc84d8